src/test/java/co/rsk/federate/FedNodeRunnerTest.java

```diff
@@ -443,69 +439,49 @@ void test_with_no_config() throws Exception {
 
     @Test
     void run_whenHsmVersionIsLowerThanThreeAndDifficultyTargetConfigIsNotPresent_shouldThrowException() throws Exception {
-        SignerConfig btcSignerConfig = getHSMBTCSignerConfig(2);
-        when(btcSignerConfig.getConfig().getString(DIFFICULTY_TARGET.getPath()))
-           .thenThrow(new ConfigException.Missing(DIFFICULTY_TARGET.getPath()));
+        int version = 2;
+        when(hsmBookkeepingClient.getVersion()).thenReturn(version);
+        when(hsmBookkeepingClient.getBlockchainParameters()).thenThrow(
+            new HSMUnsupportedTypeException("PowHSM version: " + version));
+        SignerConfig btcSignerConfig = SignerConfigBuilder.builder()
+            .withHsmSigner("m/44'/0'/0'/0/0")
+            .withValue("bookkeeping.informerInterval", 500000L)
+            .withValue("bookkeeping.maxAmountBlockHeaders", 1000)
+            .withValue("bookkeeping.maxChunkSizeToHsm", 100)
+            .withValue("bookkeeping.stopBookkeepingScheduler", true)
+            .build("BTC");
         when(fedNodeSystemProperties.signerConfig(BTC_KEY_ID.getId())).thenReturn(btcSignerConfig);
 
         assertThrows(ConfigException.class, () -> fedNodeRunner.run());
     }
 
-    private SignerConfig getBTCSignerConfig() {
-        SignerConfig btcSignerConfig = mock(SignerConfig.class);
-        when(btcSignerConfig.getId()).thenReturn("BTC");
-        when(btcSignerConfig.getType()).thenReturn("keyFile");
-        when(btcSignerConfig.getConfig()).thenReturn(keyFileConfig);
-        return btcSignerConfig;
+    private SignerConfig getBTCSignerConfig(String path) {
+        return SignerConfigBuilder.builder()
+            .withKeyFileSigner(path)
+            .build("BTC");
     }
 
-    private SignerConfig getRSKSignerConfig() {
-        SignerConfig rskSignerConfig = mock(SignerConfig.class);
-        when(rskSignerConfig.getId()).thenReturn("RSK");
-        when(rskSignerConfig.getType()).thenReturn("keyFile");
-        when(rskSignerConfig.getConfig()).thenReturn(keyFileConfig);
-        return rskSignerConfig;
+    private SignerConfig getRSKSignerConfig(String path) {
+        return SignerConfigBuilder.builder()
+            .withKeyFileSigner(path)
+            .build("RSK");
     }
 
-    private SignerConfig getMSTSignerConfig() {
-        SignerConfig mstSignerConfig = mock(SignerConfig.class);
-        when(mstSignerConfig.getId()).thenReturn("MST");
-        when(mstSignerConfig.getType()).thenReturn("keyFile");
-        when(mstSignerConfig.getConfig()).thenReturn(keyFileConfig);
-        return mstSignerConfig;
+    private SignerConfig getMSTSignerConfig(String path) {
+        return SignerConfigBuilder.builder()
+            .withKeyFileSigner(path)
+            .build("MST");
```

Comment:
We could use PowPegNodeKeyId for these keys instead of the literals

Reply:
Sounds good, I have updated it